### PR TITLE
Vue 0.12.10 and datatable refactoring/fixes

### PR DESCRIPTION
**Pull request by noirbizarre**
This pull-request upgrade Vue.js to version 0.12.10 and fixes datatable for it.

Datatable has been splitted into multiple reusable components (allows to use table outside box widget).

It fixes sort and search too.

### CODE REVIEW

File: js/components/datatable/table.vue [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    },
+    events: {
+        'datatable:item:click': function(item) {
+            this.selected = item;
+        }
+    },
+    methods: {
+        header_click: function(field) {
+            if (field.sort) {
+                this.p.sort(this.remote ? field.sort : field.key);
```
Comment (davidbgk) [inline js/components/datatable/table.vue]: Can't you use `sort_for` just below?

File: js/components/datatable/table.vue [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        return {
+            p: {},
+            track: 'id',
+            selected: null,
+            fields: []
+        };
+    },
+    computed: {
+        remote: function() {
+            return this.p && (this.p.serverside == true);
```
Comment (davidbgk) [inline js/components/datatable/table.vue]: Is the `== true` really necessary? If so, `===` is better.

File: js/components/datatable/cells/date.vue [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<template>
+<time datetime="{{value | dt YYYY-MM-DD }}">{{value | dt L }}</time>
```
Comment (davidbgk) [inline js/components/datatable/cells/date.vue]: Do we try to be consistent on spaces and mustaches?

**Comment by davidbgk** [comment]
r+wc

File: js/components/datatable/cells/date.vue (hunk 3 above)
Comment (noirbizarre) [inline js/components/datatable/cells/date.vue]: Good remark. We should write our coding guidelines somewhere.

**Pull request merged**